### PR TITLE
#94 Запланированные развлечения: пагинация

### LINKED ISSUE #94: Запланированные развлечения: пагинация, фильтрация, сортирвка
Blocked by: #72 

необходимо реализовать для странички развлечения, как на стороне апи, так и на стороне клиента следующий функционал (лучше всего именно в таком порядке):

1. Пагинация - постраничная 
2. Фильтрация:
- по стране
- по году
- по типу
3. Сортировка:
- по дате посещения
- по названию

Материалы:
- [что такое пагинация и какой она вообще бывает](https://medium.com/swlh/paginating-requests-in-apis-d4883d4c1c4c)
- [как реализовать пагинацию на серевере](https://medium.com/@zarkopafilis/asp-net-core-2-2-3-rest-api-26-pagination-650d0363ccf6) 
- [как добавить фильтрацию на сервере](https://medium.com/@zarkopafilis/asp-net-core-2-2-3-rest-api-28-resource-filtering-67fa61462c31)
 
Реализация из материалов - только возможный пример. Если хочется реализовать по-другому - велкам

На стороне API:
- [ ] модель пагинированного ответа
```
public class IPagedResponse<T>
{
    public IEnumerable<T> Data { get; set; }

    public int? PageNumber { get; set; }

    public int? PageSize { get; set; }

   public int? PageCount { get; set; }

    public int? TotalCount { get; set; }
}
```
- [ ] реализовать возможность серверной пагинации, сортировки и фильтрации, сделав необходимые изменения в контроллерах и сервисах апи.

Фронтенд
- [ ] пагинация, используем компонент вьютифай https://vuetifyjs.com/en/components/paginations/, цвет кнопок - primary цвет нашего приложения #03A9F4
- [ ] сортировка - стандарный закругленный select (тоже в #03A9F4 цвете) для выбора свойства по которому сортировать и выбор направления сортировки через группу из двух кнопок-иконок стрелок вверх-вниз цвета 
 https://vuetifyjs.com/en/components/button-groups/ цвета #03A9F4, или делаем в селекте на каждое свойство для сортировки две опции - по убыванию и по возрастанию. размещаем между кнопкой добавления развлечения и фильтром - дизайн нашего приложение, только без мультивыбора делаем  [исходник](https://www.figma.com/file/1WPfvOuZS9x7pwFlQbT1ht/%D0%9F%D1%80%D0%BE%D0%B5%D0%BA%D1%82-%D0%A2%D1%83%D1%80%D0%B8%D0%B7%D0%BC?node-id=1005%3A256) image
- [ ] фильтрация - по нажатию на кнопку фильтра - показывается как попап меню настройки фильтров, пригодятся:  
- [группировка фильтров](https://vuetifyjs.com/en/components/list-item-groups/)
- [попап меню](https://vuetifyjs.com/en/components/menus/#popover-menu) - [пример использования в дизайне ](http://joxi.ru/MAjqR3jtxPVg62)
- для выбранных фильтров можно использовать [чипсы](https://vuetifyjs.com/en/components/chip-groups/#filter-results)  - http://joxi.ru/GrqpV15CkvnW8A
- - [исходник примера дизайна ](https://dribbble.com/shots/3022156-Candidate-Listing-Page)
- - дизайн чипсов в нашем приложении [исходник](https://www.figma.com/file/1WPfvOuZS9x7pwFlQbT1ht/%D0%9F%D1%80%D0%BE%D0%B5%D0%BA%D1%82-%D0%A2%D1%83%D1%80%D0%B8%D0%B7%D0%BC?node-id=1005%3A256) image
- могут понадобиться [слайдеры ](https://vuetifyjs.com/en/components/sliders/): дизайн в приложении [исходник](https://www.figma.com/file/1WPfvOuZS9x7pwFlQbT1ht/%D0%9F%D1%80%D0%BE%D0%B5%D0%BA%D1%82-%D0%A2%D1%83%D1%80%D0%B8%D0%B7%D0%BC?node-id=1005%3A256) image
- для выбора странны селект ([мультиселект](https://vuetifyjs.com/en/components/selects/#multiple))

### PR DESCRIPTION
**Pull request by ivhorodko**
#94

### CODE REVIEW

**Comment by sanioooook** [comment]
use keywords for linked issue

File: front/src/views/profile/entertainments/entertainments.view.html [hunk 1]
```diff
@@ -43,6 +43,16 @@
           :isActive="selectedPlaceIndex && selectedPlaceIndex === index + 1 ? true : false"
```
Comment (sanioooook) [inline front/src/views/profile/entertainments/entertainments.view.html]: :isActive="selectedPlaceIndex && selectedPlaceIndex === index + 1"

result is bool, why you need return true or false?

File: front/src/views/profile/entertainments/entertainments.view.js [hunk 2]
```diff
@@ -27,14 +28,15 @@ export default {
...
-      markerPos: null
+      markerPos: null,
+      pageNumber: 1,
+      totalEntertainmentsPagesCount: 1
     };
   },
   async mounted () {
     await this.initMap();
-    await this.$store.dispatch('entertainment/initEntertainments');
+    await this.getEntertainments(0);
```
Comment (sanioooook) [inline front/src/views/profile/entertainments/entertainments.view.js:38]: do you really need sync-await?

File: server/WebAPI/DTO/PagedResponse.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  {
+    public IEnumerable<T> Data { get; set; }
+
+    public int? PageNumber { get; set; }
+
+    public int? PageSize { get; set; }
+
+    public int? TotalCount { get; set; }
+  }
+}
```
Comment (sanioooook) [inline server/WebAPI/DTO/PagedResponse.cs]: new line

File: server/WebAPI/DTO/PaginationRequestDTO.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+{
+  public class PaginationRequestDTO
+  {
+
+    [Required]
+    [Range(0, Int32.MaxValue)]
+    public int? PageNumber { get; set; }
+
+    [Required]
+    [Range(1, 24)]
```
Comment (sanioooook) [inline server/WebAPI/DTO/PaginationRequestDTO.cs:14]: why not int.max? why 24?)

Comment (ivhorodko) [reply server/WebAPI/DTO/PaginationRequestDTO.cs:14]: cause i don't need to get 2 147 483 647 results for one time

Comment (valerachuk) [reply server/WebAPI/DTO/PaginationRequestDTO.cs:14]: 24 - magic number )

Comment (ivhorodko) [reply server/WebAPI/DTO/PaginationRequestDTO.cs:14]: Well, I consider that 24 is enough for the maximum number of results

**Review by sanioooook — APPROVED** [review]

**Pull request merged**